BetaCameras/Xtion2/Xtion2.cs

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) Metrilus GmbH
+// MetriCam 2 is licensed under the MIT license. See License.txt for full license text.
+
+using Metrilus.Util;
+using System;
+using System.Drawing;
+using System.Drawing.Imaging;
+
+
+namespace MetriCam2.Cameras
+{
+    public class Xtion2 : Camera, IDisposable
+    {
+        private bool _disposed = false;
+        OpenNICAPI.Device _device = new OpenNICAPI.Device();
+        OpenNICAPI.Stream _colorStream = new OpenNICAPI.Stream();
+        OpenNICAPI.Stream _depthStream = new OpenNICAPI.Stream();
+        OpenNICAPI.Stream _irStream = new OpenNICAPI.Stream();
+        OpenNICAPI.Frame _colorFrame = new OpenNICAPI.Frame();
+        OpenNICAPI.Frame _depthFrame = new OpenNICAPI.Frame();
+        OpenNICAPI.Frame _irFrame = new OpenNICAPI.Frame();
+
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed)
+                return;
+
+            OpenNICAPI.Shutdown();
+            _disposed = true;
+        }
+
+        #region Constructor
+        public Xtion2()
+        {
+            OpenNICAPI.Init();
+        }
+
+        ~Xtion2()
+        {
+            Dispose(false);
+        }
+        #endregion
+
+        #region MetriCam2 Camera Interface Methods
+
+        protected override void LoadAllAvailableChannels()
+        {
+            ChannelRegistry cr = ChannelRegistry.Instance;
+            Channels.Clear();
+
+            Channels.Add(cr.RegisterChannel(ChannelNames.ZImage));
+            Channels.Add(cr.RegisterChannel(ChannelNames.Color));
+			Channels.Add(cr.RegisterChannel(ChannelNames.Intensity));
+        }
+
+
+        protected override void ConnectImpl()
+        {
+            _device = OpenNICAPI.OpenDevice(this.SerialNumber);
+            _colorStream = OpenNICAPI.CreateStream(_device, OpenNICAPI.OniSensorType.COLOR);
+            _depthStream = OpenNICAPI.CreateStream(_device, OpenNICAPI.OniSensorType.DEPTH);
+            _irStream = OpenNICAPI.CreateStream(_device, OpenNICAPI.OniSensorType.IR);
+
+            ActivateChannel(ChannelNames.Color);
```

Comment:
This will overrule any channel's a user has set before calling `Connect`.
It's ok for now, but once I have written a test case for it you will have to fix it ;-)

Reply:
Is it okay to activate a channel in `LoadAllAvailableChannels`? Otherwise all channels will be deactivated by default.